src/components/presets/account_preset.cairo

```diff
@@ -90,10 +132,41 @@ pub mod AccountPreset {
     #[abi(embed_v0)]
     impl Lockable of ILockable<ContractState> {
         fn lock(ref self: ContractState, lock_until: u64) {
+            // validate signer
+            let caller = get_caller_address();
+            assert(self.is_valid_signer(caller), 'Account: unauthorized');
+
+            // lock account
             self.lockable.lock(lock_until);
         }
+
         fn is_locked(self: @ContractState) -> (bool, u64) {
             self.lockable.is_locked()
         }
     }
+
+    // *************************************************************************
+    //                              PERMISSIONABLE IMPL
+    // *************************************************************************
+    #[abi(embed_v0)]
+    impl Permissionable of IPermissionable<ContractState> {
+        fn set_permission(
+            ref self: ContractState,
+            permissioned_addresses: Array<ContractAddress>,
+            permissions: Array<bool>
+        ) {
+            // validate signer
+            let caller = get_caller_address();
+            assert(self.is_valid_signer(caller), 'Account: unauthorized');
```

Comment:
folllowing my earlier comment on enforcing only owner to call the `set_permission` function, this check for valid signer becomes unnecessary and redundant. Let's take it off

Reply:
this hasn't been resolved too